mitiq/executor/executor.py

```diff
@@ -84,10 +91,34 @@ def __init__(
 
     @property
     def can_batch(self) -> bool:
-        return self._executor_return_type in (
+        """Returns True if the executor is recognized as a "batched executor",
+        else False.
+
+        The executor is detected as "batched" if and only if it is annotated
+        with a return type that is a subclass of ``Iterable``. Common examples
+        include:
+
+        * ``Iterable[QuantumResult]``
+        * ``List[QuantumResult]``/``list[QuantumResult]``
+        * ``Sequence[QuantumResult]``
+        * ``Tuple[QuantumResult]``/``tuple[QuantumResult]``
+
+        Otherwise, it is considered "serial".
+
+        Batched executors can *run several quantum programs in a single call*.
+
+        Returns:
+            True if the executor is detected as batched, else False.
+        """
+        return_type = self._executor_return_type
+
+        if return_type is None:
+            return False
+
+        return return_type in (
             BatchedType[T]  # type: ignore[index]
-            for BatchedType in [Iterable, List, Sequence, Tuple]
-            for T in QuantumResult.__args__  # type: ignore[attr-defined]
+            for BatchedType in [Iterable, List, Sequence, Tuple, list, tuple]
```

Comment:
`typing.Sequence` is also [deprecated](https://docs.python.org/3/library/typing.html#typing.Sequence) in favor of `collections.abc.Sequence`

Reply:
Actually I think all the data types in `typing` are deprecated and have been moved to `collections.abc`. I've opened https://github.com/unitaryfund/mitiq/issues/2701 to ensure we upgrade, but I think that is best done in a separate PR.

Reply:
What I meant is that how we added `list`, and `tuple`, which are the non deprecated alternative of `List` and `Tuple`, then we should add `abc.Sequence` as a non-deprecated alternative for `typing.Sequence`.

Reply:
I see. Added an additional type in https://github.com/unitaryfund/mitiq/pull/2700/commits/d2ba012f13b0bb43e361a9c0a251d071df1f5e4f.

This made me realize that if someone typehints an executor with `collections.abc.Iterable`, we will misclassify it as a serial executor since `typing.Iterable != collections.abc.Iterable`. Ugh.

Reply:
The bottom-line problem is this use of type checks at runtime. In the next Mitiq Coding session, let's brainstorm if/why we need those at all.